### PR TITLE
Add menu item and link send page to utxo page

### LINKED ISSUE #588: link custom input to the send page
Custom input has already been implemented and can be seen here https://github.com/planetdecred/godcr/blob/master/ui/page/utxo_page.go. This page should be linked to the send page. An "advanced mode" should be added similar to the mobile mockup below.
  
Screenshot 2021-09-01 at 20 06 49

Clicking the "advanced mode" option should lead to the custom inputs page.

### PR DESCRIPTION
**Pull request by JustinBeBoy**
Resolve #588 
This PR link send page to utxo page

image

### CODE REVIEW

File: ui/page/components/utils.go [hunk 1]
```diff
@@ -12,3 +16,10 @@ func ContextDone(ctx context.Context) bool {
 		return false
 	}
 }
+
+func MustIcon(ic *widget.Icon, err error) *widget.Icon {
+	if err != nil {
+		panic(err)
+	}
+	return ic
+}
```
Comment (Sirmorrison) [inline ui/page/components/utils.go]: This already exist on the `theme.go` file but not exported. 

This method is also repeating across multiple files (links below).
Kindly remove all instances of this method and make use of the method you implemented on the components >> utils.go file.

other instance of the `mustIcon` methods can be found here.
https://github.com/planetdecred/godcr/blob/5603d402151a1ecad15cec83dbc2d041dc50c1bd/ui/util.go#L35

https://github.com/planetdecred/godcr/blob/5603d402151a1ecad15cec83dbc2d041dc50c1bd/ui/load/utils.go#L10

https://github.com/planetdecred/godcr/blob/5603d402151a1ecad15cec83dbc2d041dc50c1bd/ui/page/utils.go#L34

https://github.com/planetdecred/godcr/blob/5603d402151a1ecad15cec83dbc2d041dc50c1bd/ui/decredmaterial/theme.go#L226

File: ui/page/send/layout.go [hunk 2]
```diff
@@ -5,7 +5,9 @@ import (
 
 	"gioui.org/layout"
 	"gioui.org/op"
+	"gioui.org/unit"
 	"gioui.org/widget"
+	"gioui.org/widget/material"
```
Comment (Sirmorrison) [inline ui/page/send/layout.go]: Not needed... we already have `decredmaterial.Clickable()`

File: ui/page/send/layout.go [hunk 3]
```diff
@@ -90,6 +82,56 @@ func (pg *Page) topNav(gtx layout.Context) layout.Dimensions {
...
+	m := op.Record(gtx.Ops)
+	inset.Layout(gtx, func(gtx C) D {
+		width := unit.Value{U: unit.UnitDp, V: 150}
+		gtx.Constraints.Max.X = gtx.Px(width)
+		return pg.shadowBox.Layout(gtx, func(gtx C) D {
+			return pg.optionsMenuCard.Layout(gtx, func(gtx C) D {
+				return (&layout.List{Axis: layout.Vertical}).Layout(gtx, len(pg.moreItems), func(gtx C, i int) D {
+					return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
+						layout.Rigid(func(gtx C) D {
+							return material.Clickable(gtx, pg.moreItems[i].button, func(gtx C) D {
```
Comment (Sirmorrison) [inline ui/page/send/layout.go]:
```suggestion
							return decredmaterial.Clickable(gtx, pg.moreItems[i].button, func(gtx C) D {
```

File: ui/page/send/layout.go [hunk 4]
```diff
@@ -90,6 +82,56 @@ func (pg *Page) topNav(gtx layout.Context) layout.Dimensions {
...
+								m10 := values.MarginPadding10
+								return layout.Inset{Top: m10, Bottom: m10, Left: m10, Right: m10}.Layout(gtx, func(gtx C) D {
```
Comment (Sirmorrison) [inline ui/page/send/layout.go]:
```suggestion
								return layout.UniformInset(values.MarginPadding10).Layout(gtx, func(gtx C) D {
```

File: ui/page/send/page.go [hunk 5]
```diff
@@ -5,6 +5,8 @@ import (
 	"strconv"
 	"strings"
 
+	"gioui.org/widget"
+
```
Comment (Sirmorrison) [inline ui/page/send/page.go]: remove white space

File: ui/page/send/page.go [hunk 6]
```diff
@@ -318,14 +338,28 @@ func (pg *Page) Handle() {
...
+}
 
-		pg.amount.clearAmount()
+func (pg *Page) mustIcon(ic *widget.Icon, err error) *widget.Icon {
+	if err != nil {
+		panic(err)
 	}
-
+	return ic
 }
```
Comment (Sirmorrison) [inline ui/page/send/page.go]: There are multiple instance of this already... Kindly remove and use on from the either the component/utils.go 

See comments above.

File: ui/page/send/utxo_page.go [hunk 7]
```diff
@@ -136,12 +138,12 @@ func (pg *UTXOPage) calculateAmountAndFeeUTXO() {
 	}
 	err := pg.txAuthor.UseInputs(utxoKeys)
 	if err != nil {
-		log.Error(err)
+		//pg.log.Error(err)
```
Comment (Sirmorrison) [inline ui/page/send/utxo_page.go]: Why was this commented out?

File: ui/page/send/utxo_page.go [hunk 8]
```diff
@@ -136,12 +138,12 @@ func (pg *UTXOPage) calculateAmountAndFeeUTXO() {
...
 		return
 	}
 	feeAndSize, err := pg.txAuthor.EstimateFeeAndSize()
 	if err != nil {
-		log.Error(err)
+		//pg.log.Error(err)
```
Comment (Sirmorrison) [inline ui/page/send/utxo_page.go]: This also.

File: ui/page/send/utxo_page.go [hunk 9]
```diff
@@ -156,56 +158,56 @@ func (pg *UTXOPage) clearPageData() {
...
 								}),
-								layout.Flexed(0.25, func(gtx C) D {
+								layout.Flexed(0.25, func(gtx layout.Context) layout.Dimensions {
 									return pg.textData(gtx, "Fee:  ", pg.txnFee)
 								}),
-								layout.Flexed(0.25, func(gtx C) D {
+								layout.Flexed(0.25, func(gtx layout.Context) layout.Dimensions {
 									return pg.textData(gtx, "After Fee:  ", pg.txnAmountAfterFee)
 								}),
 							)
```
Comment (Sirmorrison) [inline ui/page/send/utxo_page.go]: Why was all these changed?

File: ui/values/localizable/en.go [hunk 10]
```diff
@@ -117,4 +117,6 @@ const EN = `
 "usdBittrex" = "USD (Bittrex)";
 "none" = "None";
 "proposals" = "Proposals";
+"advancedMode" = "Advanced mode"
+"clearAllFields" = "Clear all fields"
```
Comment (Sirmorrison) [inline ui/values/localizable/en.go]: Please revert this... Use the strings value locally.

File: ui/values/strings.go [hunk 11]
```diff
@@ -223,4 +223,6 @@ const (
 	StrUsdBittrex                  = "usdBittrex"
 	StrNone                        = "none"
 	StrProposal                    = "proposals"
+	StrAdvancedMode                = "advancedMode"
+	StrClearFeilds                 = "clearAllFields"
```
Comment (Sirmorrison) [inline ui/values/strings.go]: Revert this also. Use the strings locally. 
The localized strings file is not to be edited.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

File: ui/page/send/page.go [hunk 12]
```diff
@@ -115,6 +129,11 @@ func NewSendPage(l *load.Load) *Page {
 
 	pg.initLayoutWidgets()
 
+	pg.optionsMenuCard = decredmaterial.Card{Color: pg.Theme.Color.Surface}
+	pg.optionsMenuCard.Radius = decredmaterial.Radius(5)
+
+	pg.moreItems = pg.getMoreItem()
+
```
Comment (beansgum) [inline ui/page/send/page.go]: Put this in initLayoutWidgets() function.

File: ui/page/send/page.go [hunk 13]
```diff
@@ -318,12 +337,19 @@ func (pg *Page) Handle() {
...
 
-	for pg.clearAllBtn.Clicked() {
-		pg.moreOptionIsOpen = false
-
-		pg.sendDestination.clearAddressInput()
+	for _, menu := range pg.moreItems {
+		if menu.button.Clicked() {
+			switch menu.id {
+			case UTXOPageID:
+				pg.ChangeFragment(NewUTXOPage(pg.Load))
```
Comment (beansgum) [inline ui/page/send/page.go]: This can be in the action function of the moreItem.

File: ui/page/send/layout.go [hunk 14]
```diff
@@ -90,6 +81,55 @@ func (pg *Page) topNav(gtx layout.Context) layout.Dimensions {
...
+			button: new(widget.Clickable),
+			id:     UTXOPageID,
+		},
+		{
+			text:   "Clear all fields",
+			button: new(widget.Clickable),
+			action: func() {
+				pg.moreOptionIsOpen = false
+				pg.sendDestination.clearAddressInput()
+				pg.amount.clearAmount()
```
Comment (beansgum) [inline ui/page/send/layout.go]: Use `resetFields()`

**Review by beansgum — CHANGES_REQUESTED** [review]

File: ui/page/components/utils.go [hunk 15]
```diff
@@ -1,6 +1,8 @@
 package components
 
-import "context"
+import (
+	"context"
+)
```
Comment (Sirmorrison) [inline ui/page/components/utils.go]: Please revert this.

File: ui/page/send/layout.go [hunk 16]
```diff
@@ -5,6 +5,7 @@ import (
 
 	"gioui.org/layout"
 	"gioui.org/op"
+	"gioui.org/unit"
```
Comment (Sirmorrison) [inline ui/page/send/layout.go]: Remove this.. Use the values defined in the Dimensions.go file.

File: ui/page/send/layout.go [hunk 17]
```diff
@@ -90,6 +84,57 @@ func (pg *Page) topNav(gtx layout.Context) layout.Dimensions {
...
+				pg.moreOptionIsOpen = false
+			},
+		},
+	}
+}
+
+func (pg *Page) layoutOptionsMenu(gtx layout.Context) {
+	inset := layout.Inset{
+		Top:  unit.Dp(30),
+		Left: unit.Dp(-120),
```
Comment (Sirmorrison) [inline ui/page/send/layout.go]:
```suggestion
		Top: values.MarginPadding30,
		Left: values.MarginPaddingMinus120,
```
Add them to the Dimensions.go file if they dont exist.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
image

app crash with the following error.

**Review by beansgum — APPROVED** [review]

**Review by dreacot — APPROVED** [review]

**Review by Sirmorrison — APPROVED** [review]

**Pull request merged**